lab_1/main.py

```diff
@@ -98,7 +199,31 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
+
+    score = {}
+    if not (isinstance(unknown_profile, dict) and isinstance(profiles, list)
+            and isinstance(languages, list) and isinstance(top_n, int)):
+        return None
+    exist_lang = False
+    for _ in profiles:
+        if languages:
+            for i in languages:
+                if i == _.get('name'):
+                    exist_lang = True
+                    score[i] = compare_profiles_advanced(unknown_profile, _, top_n).get('score')
+        else:
+            score[_.get('name')] = compare_profiles_advanced(unknown_profile, _, top_n).get('score')
+    if not exist_lang and languages:
+        return None
+    sorted_score = sorted(score.items(), key=lambda x: x[1], reverse=True)
+    i = 0
+    first_score = sorted_score[0][1]
+    name_list = []
+    while i < len(sorted_score) and first_score == sorted_score[i][1]:
+        name_list.append(sorted_score[i][0])
+        i += 1
+    name_list_sorted = sorted(name_list)
```

Comment:
explain what happens here, please
a bit hard to read

```
 sorted_score = sorted(score.items(), key=lambda x: x[1], reverse=True)
     i = 0
     first_score = sorted_score[0][1]
     name_list = []
     while i < len(sorted_score) and first_score == sorted_score[i][1]:
         name_list.append(sorted_score[i][0])
         i += 1
     name_list_sorted = sorted(name_list)
```

Reply:
sorry i don't remember how i wrote it, i changed it just now, so it has to be easier to understand. 
firstly, it sorts the dict by the largest score, then it takes the first one and checks if there's another score that is the same, then it puts the languages with the same score in a list and sorts it in alphabetical order